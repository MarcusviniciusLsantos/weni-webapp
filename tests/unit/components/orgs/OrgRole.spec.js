import { shallowMount, createLocalVue } from '@vue/test-utils';
import OrgRole from '@/components/orgs/orgRole.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('OrgRole.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(OrgRole, {
      localVue,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicButton: true,
        UnnnicDropdown: true,
        UnnnicDropdownItem: true,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('roleOptions()', async () => {
    await wrapper.setData({
      roles: {
        1: 'view',
        2: 'contributor',
        3: 'admin',
      },
    });
    const response = wrapper.vm.roleOptions;
    expect(response).toStrictEqual(Object.keys(wrapper.vm.roles));
  });

  it('labelFor()', async () => {
    await wrapper.setData({
      roles: {
        1: 'view',
        2: 'contributor',
        3: 'admin',
      },
    });
    const response = wrapper.vm.labelFor(1);
    expect(response).toStrictEqual(wrapper.vm.$t());
  });

  it('onSelectRole()', async () => {
    const param = 'test';
    wrapper.vm.onSelectRole(param);
    expect(wrapper.vm.currentRole).toBe(param);
    expect(wrapper.emitted('onChangeRole')).toBeTruthy();
  });

  it('onDelete()', async () => {
    wrapper.vm.onDelete();
    expect(wrapper.emitted('onDelete')).toBeTruthy();
  });
});
