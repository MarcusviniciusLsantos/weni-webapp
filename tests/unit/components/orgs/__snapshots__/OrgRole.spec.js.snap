// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgRole.vue renders a snapshot 1`] = `
<div class="weni-org-role">
  <avatar-stub size="sm"></avatar-stub>
  <div class="weni-org-role__info">
    <p class="weni-org-role__info__name"></p>
    <p class="weni-org-role__info__email"></p>
  </div>
  <div class="weni-org-role__role">
    <!---->
    <unnnicdropdown-stub>
      <unnnicbutton-stub size="small" type="terciary" icon-right="arrow-down-1-1" class="weni-org-role__action__button">
        some specific text
      </unnnicbutton-stub>
      <unnnicdropdownitem-stub>
        some specific text
      </unnnicdropdownitem-stub>
      <unnnicdropdownitem-stub>
        some specific text
      </unnnicdropdownitem-stub>
      <unnnicdropdownitem-stub>
        some specific text
      </unnnicdropdownitem-stub>
    </unnnicdropdown-stub>
    <!---->
  </div>
</div>
`;
